rskj-core/src/main/java/co/rsk/net/SyncProcessor.java

```diff
@@ -0,0 +1,319 @@
+package co.rsk.net;
+
+import co.rsk.core.bc.BlockChainStatus;
+import co.rsk.net.messages.*;
+import co.rsk.net.sync.*;
+import co.rsk.validators.BlockHeaderValidationRule;
+import com.google.common.annotations.VisibleForTesting;
+import org.ethereum.core.Block;
+import org.ethereum.core.BlockHeader;
+import org.ethereum.core.BlockIdentifier;
+import org.ethereum.core.Blockchain;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.util.ByteUtil;
+import org.ethereum.validator.DependentBlockHeaderRule;
+import org.ethereum.validator.DifficultyRule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.*;
+
+/**
+ * Created by ajlopez on 29/08/2017.
+ * This class' methods are executed one at a time because NodeMessageHandler is synchronized.

```

Comment:
We can improve this

Reply:
Or add it to the backlog

Reply:
Yes we can improve it on later release :+1: